internal/server/data/group.go

```diff
@@ -33,48 +31,113 @@ func CreateGroup(tx WriteTxn, group *models.Group) error {
 	return insert(tx, (*groupsTable)(group))
 }
 
-func GetGroup(db GormTxn, selectors ...SelectorFunc) (*models.Group, error) {
-	group, err := get[models.Group](db, selectors...)
+type GetGroupOptions struct {
+	// ByID instructs GetGroup to return the group matching this ID.
+	ByID uid.ID
+	// ByName instructs GetGroup to return the group matching this name.
+	ByName string
+}
+
+func GetGroup(tx ReadTxn, opts GetGroupOptions) (*models.Group, error) {
+	group := &groupsTable{}
+	query := querybuilder.New("SELECT")
+	query.B(columnsForSelect(group))
+	query.B("FROM groups")
+	query.B("WHERE deleted_at is null")
+	query.B("AND organization_id = ?", tx.OrganizationID())
+
+	switch {
+	case opts.ByID != 0:
+		query.B("AND id = ?", opts.ByID)
+	case opts.ByName != "":
+		query.B("AND name = ?", opts.ByName)
+	default:
+		return nil, fmt.Errorf("GetGroup requires an ID")
+	}
+
+	err := tx.QueryRow(query.String(), query.Args...).Scan(group.ScanFields()...)
 	if err != nil {
-		return nil, err
+		return nil, handleReadError(err)
 	}
 
-	count, err := CountUsersInGroup(db, group.ID)
+	count, err := countUsersInGroup(tx, group.ID)
 	if err != nil {
 		return nil, err
 	}
 	group.TotalUsers = int(count)
-	return group, nil
+	return (*models.Group)(group), nil
 }
 
-func ListGroups(db GormTxn, p *Pagination, selectors ...SelectorFunc) ([]models.Group, error) {
-	groups, err := list[models.Group](db, p, selectors...)
+type ListGroupsOptions struct {
+	ByName string
+
+	// ByGroupMember instructs ListGroups to return groups where this user ID
+	// is a member of the group.
+	ByGroupMember uid.ID
+
+	Pagination *Pagination
+}
 
+func ListGroups(tx ReadTxn, opts ListGroupsOptions) ([]models.Group, error) {
+	table := groupsTable{}
+	query := querybuilder.New("SELECT")
+	query.B(columnsForSelect(table))
+	if opts.Pagination != nil {
+		query.B(", count(*) OVER()")
+	}
+	query.B("FROM groups")
+	if opts.ByGroupMember != 0 {
+		query.B("JOIN identities_groups ON groups.id = identities_groups.group_id")
+		query.B("AND identities_groups.identity_id = ?", opts.ByGroupMember)
```

Comment:
Shouldn't the group member identity ID be a where clause rather than part of the join?

Reply:
Ok, I think I found an answers to this: https://dba.stackexchange.com/a/155976

It sounds like for inner joins (`JOIN` by itself is an inner join [ref](https://www.postgresql.org/docs/current/queries-table-expressions.html#QUERIES-JOIN)) it makes no difference to postgreSQL. It will perform the same query.

For us, the advantage to doing it here in the `JOIN` is that we don't need a second `if opts.ByGroupMember`. We can include all of the clauses in the same code block. If we were to try and use the where we'd have to split it up.